tutorials/plot_background_morph.py

```diff
@@ -0,0 +1,875 @@
+the representation of spatio-temporal activation patterns is often achieved by
+overlaying the anatomical brain structure, with the respective activation at
+the respective anatomical location. Hereby volumetric anatomical MR images are
+often used as such or are transformed into an inflated surface.
+
+It becomes obvious that in order to compute group level statistics, data
+representations across subjects must be morphed to a common frame, such that
+anatomically and functional similar structures are represented at the same
+spatial location.
+
+.. _tut_morphing_basics:
+
+Morphing basics
+================
+
+Morphing describes the procedure of transforming a data representation in n-
+dimensional space into another data representation in the same space. In the
+context of neuroimaging data this space will mostly be 3-dimensional data and
+is necessary to bring individual subject brain spaces into a common frame. Is
+it possible to just shift one brain image such that it is overlaying with a
+second?
+
+Generally speaking yes, but it needs to be kept in mind, that in order to
+achieve a highly accurate overlay, morphological differences have to be
+accounted as well. Not just do sizes of brains vary, but also how cortical and
+subcortical structures are expressed in terms of size and their location
+relative to other structures. Thus it is inevitable to use slightly more
+complex operations than just shifting brains relative to each other.
+Nevertheless, this can be a first step of a successful subject to subject
+transformation.
+
+We can understand mapping as a mathematical function :math:`f`. This function
+can be seen as a description on how to move from one point in n-dimensional
+space to another. Like with a real maps, when attempting to reach a new real
+world location, we try to derive a set of actions (in the example directional
+decisions to reach the new location), to move from one point to another. This
+set of actions can also be described as a function of the current position
+that computes the new location.
+
+In general morphing operations can be split into two different kinds of
+transformation: linear and non-linear morphs or mappings:
+
+A mapping is linear if it satisfies the following two conditions:
+
+.. math::    f(u + v) = f(u) + f(v)\ ,
+.. math::    f(cu) = cf(u)\ ,
+
+where :math:`u` and :math:`v` are from the same vector space and :math:`c` can
+be any scalar. This means that any linear transform can be described as a
+mixture of additive and multiplicative operations and hence is often
+represented in terms of a transformation matrix:
+
+.. math::
+
+    x^{(B)}=
+    \begin{bmatrix}
+    x\prime \\
+    y\prime \\
+    z\prime \\
+    1 \end{bmatrix} =
+    M^{(AB)}x^{(A)}=
+    \begin{bmatrix}
+    1 & 0 & 0 & 0 \\
+    0 & 1 & 0 & 0 \\
+    0 & 0 & 1 & 0 \\
+    0 & 0 & 0 & 1
+    \end{bmatrix}
+    \begin{bmatrix}
+    x \\
+    y \\
+    z \\
+    1
+    \end{bmatrix}\ ,
```

Comment:
does this mean `M^{(AB)} = I_4` ?

Reply:
Err... yes... if I_4 means identity matrix with 4 columns then yes... but why? Shall I indicate that? I just used the identity matrix when I wrote it out to get a visual depiction of some transformation matrix. It also nice to make this easy example with scaling, because it becomes immediately obvious how transformation matrices work (I think ;) )